addons/hide-stage/userscript.js

```diff
@@ -20,38 +25,56 @@ export default async function ({ addon, console, msg }) {
     if (!bodyWrapper) return;
     document.body.classList.remove("sa-stage-hidden-outer");
     bodyWrapper.classList.remove("sa-stage-hidden");
-    hideStageButton.classList.add(addon.tab.scratchClass("stage-header_stage-button-toggled-off"));
+    // Temporary code to support both current and future Scratch version
+    if (addon.tab.scratchClass("stage-header_stage-button-toggled-off"))
+      hideStageButton.classList.add(addon.tab.scratchClass("stage-header_stage-button-toggled-off"));
+    hideStageButton.ariaPressed = false;
+    if (e) {
+      const clickedButton = e.target.closest("button");
+      if (clickedButton) clickedButton.ariaPressed = true;
+    } else if (addon.tab.redux.state) {
+      if (addon.tab.redux.state.scratchGui.stageSize.stageSize === "small") smallStageButton.ariaPressed = true;
+      else largeStageButton.ariaPressed = true;
+    }
     window.dispatchEvent(new Event("resize")); // resizes the code area and paint editor canvas
   }
 
   addon.self.addEventListener("disabled", () => unhideStage());
 
   while (true) {
-    const stageControls = await addon.tab.waitForElement("[class*='stage-header_stage-size-toggle-group_']", {
+    let stageControls = await addon.tab.waitForElement("[class*='stage-header_stage-size-toggle-group_']", {
       markAsSeen: true,
       reduxCondition: (state) => !state.scratchGui.mode.isPlayerOnly,
     });
+    // Temporary code to support both current and future Scratch version
```

Comment:
I assume we'll need to do this in many more places, it might be better to create a dedicated property or `addon.tab` or a module that can be imported.
Possibly a class name can be added to the body tag as well. What do you think?

Reply:
Let me clarify: I was wondering if a boolean property/method would be useful to conditionally run code depending on whether Scratch is running in the old version or the new version.